### DISCUSSION TITLE
LAB TIFF types

### DISCUSSION THREAD

**Discussion by SylvainArd** [ImageMagick/ImageMagick#5065] [Help]
Hello all !
I want to know with image magick what is the LAB encoding  of a LAB TIFF (CIELab, ICCLab and JPEG Fax) and its QuantumRange please.
Thank you
Best regards

**Comment by SylvainArd** [comment]
and if imagemagick respects this standard https://developer.adobe.com/content/dam/udp/en/open/standards/tiff/TIFFphotoshop.pdf page 12

**Reply by fmw42** [reply]
ImageMagick used the libtiff delegate. Consult the libtiff delegate for information about its standard and how it might relate to photoshop.

**Comment by SylvainArd** [comment]
OK thank you
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers


Le jeu. 21 avr. 2022 à 18:03, Fred Weinhaus ***@***.***> a
écrit :

**Comment by SylvainArd** [comment]
the libtiff forum seems to not work, do you know one please ?
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Reply by fmw42** [reply]
http://www.libtiff.org

**Comment by snibgo** [comment]
> and its QuantumRange please.

QuantumRange is a characteristic of a build of ImageMagick, not of file formats. You can find the value for your version of IM like this:
```
f:\web\im>%IMG7%magick xc: -format %[fx:QuantumRange] info:
65535

f:\web\im>%FXNEW%magick xc: -format %[fx:QuantumRange] info:
4.2949673e+09
```
Notice that I have shown two different versions of IM. The first is Q16, the second is Q32.

**Comment by SylvainArd** [comment]
OK so all I have to do for retrieve the values of CieLab of ranges 0-100
for L, and -128 - 127 for A and B is :
for L : 100* value/quantumrange
for A and B : (256*value/quantumrange)-128
?
because the values are not RAW TIFF values but IM specific values ?
and values goes from 0 to quantumrange or from 0 to quantumrange -1 ?
thank you very much !
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by SylvainArd** [comment]
yes I have already seen this website, but they have no working  forum or
mailing list
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers


Le jeu. 21 avr. 2022 à 18:51, Fred Weinhaus ***@***.***> a
écrit :

**Reply by fmw42** [reply]
It shows:

Mailing List: [tiff@remotesensing.org](mailto:tiff@remotesensing.org), [Subscription](http://remotesensing.org/mailman/listinfo/tiff/), [Archive](http://www.awaresystems.be/imaging/tiff/tml.html)

**Comment by SylvainArd** [comment]
Hello I had no answer for my last question :
OK so all I have to do for retrieve the values of CieLab of ranges 0-100
for L, and -128 - 127 for A and B is :
for L : 100* value/quantumrange
for A and B : (256*value/quantumrange)-128
?
because the values are not RAW TIFF values but IM specific values ?
and values goes from 0 to quantumrange or from 0 to quantumrange -1 ?
thank you very much !

and for the unavailability of the libtiff mailing list

thank you
best regards
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers

**Comment by snibgo** [comment]
I answered that in your other post, https://github.com/ImageMagick/ImageMagick/discussions/5059

Nominal values are in the range 0 to QuantumRange. If you have Lab values converted by profile, the a* and b* channels are in the nominal range -QuantumRange/2 to +QuantumRange/2.

To change the range, divide by the old maximum (eg QuantumRange) and multiply by the new maximum (eg 100 or 255). Windows BAT command:
```
magick ^
  in.tiff ^
  -channel 0 -evaluate Multiply %%[fx:100/QuantumRange] ^
  -channel 1,2 -evaluate Multiply %%[fx:255/QuantumRange] ^
  +channel ^
  ftxt:
```

I know nothing about any libtiff mailing list.

**Comment by SylvainArd** [comment]
Ok thank you very much

Envoyé de mon iPhone

**Comment by SylvainArd** [comment]
click on subscription and you will see the problem
Sylvain Ard
0549507724
0778380991
***@***.***
http://sylvain-ard.fr
Entreprise individuelle SIRET : 80079243400022
Appt 26 Bât A Résidence Le Patio
83 rue de la Bugellerie
86000 Poitiers


Le jeu. 21 avr. 2022 à 19:57, Fred Weinhaus ***@***.***> a
écrit :

**Comment by blue-j** [comment]
Photoshop equates Lab color with D50 Lab, while it's more than possible to have a D65 white point.  And believe it or not, the common -127 through 128 range for a* and b* is an _artificial clamp_ put on for programming ease, not part of the CIELAB spec either.

Why not just make test TIFFs with the latest IM ands libtiff and see if they make what you want?  TIFF is a very flexible format, accommodating varying bit depths and white points.

- J

**Comment by blue-j** [comment]
Maybe this page will help you?

http://www.libtiff.org/support.html
